x/emissions/module/rewards_v2.go

```diff
@@ -0,0 +1,56 @@
+package module
+
+import (
+	"fmt"
+	"math"
+)
+
+// GetfUniqueAgg calculates the unique value or impact of each forecaster.
+// f^+
+func GetfUniqueAgg(numForecasters float64) float64 {
+	return  1.0 / math.Pow(2.0, (numForecasters - 1.0))
+}
+
+// GetFinalWorkerScoreForecastTask calculates the worker score in forecast task.
+// T_ik
+func GetFinalWorkerScoreForecastTask(scoreOneIn, scoreOneOut, fUniqueAgg float64) float64 {
+	return fUniqueAgg*scoreOneIn + (1-fUniqueAgg)*scoreOneOut
+}
+
+// GetWorkerScore calculates the worker score based on the losses and lossesCut.
+// Consider the staked weighted inference loss and one-out loss to calculate the worker score.
+// T_ij / T^-_ik / T^+_ik
+func GetWorkerScore(losses, lossesOneOut float64) float64 {
+	deltaLogLoss := math.Log10(lossesOneOut) - math.Log10(losses)
+	return deltaLogLoss
+}
+
+// GetStakeWeightedLoss calculates the stake-weighted average loss.
+// Consider the losses and the stake of each reputer to calculate the stake-weighted loss.
+// The stake weighted loss is used to calculate the network-wide losses.
+// L_i / L_ij / L_ik / L^-_i / L^-_il / L^+_ik
+func GetStakeWeightedLoss(reputersStakes, reputersReportedLosses []float64) (float64, error) {
+	if len(reputersStakes) != len(reputersReportedLosses) {
+		return 0, fmt.Errorf("slices must have the same length")
```

Comment:
excuse me ser these should all be cosmos errors living in error.go

Reply:
will do ser! was finishing the other formulas to batch all the errors and address them in one go